pysgrid/utils.py

```diff
@@ -207,9 +207,116 @@ def calculate_angle_from_true_east(lon_lat_1, lon_lat_2):
+                    'v2rho': np.array([[0, 0], [1, 0], [0, -1], [1, -1], [0, 1], [1, 1]]),
+                    'v2psi': np.array([[0, -1], [0, 0], [-1, -1], [-1, 0], [1, -1], [1, 0]]),
+                    }
+    translations.update({'rho2psi': -translations['psi2rho'],
+                         'v2u': -translations['u2v'],
+                         'rho2u': -translations['u2rho'],
+                         'psi2u': -translations['u2psi'],
+                         'rho2v': -translations['v2rho'],
+                         'psi2v': -translations['v2psi']
+                         })
+    if translation is None or translation not in translations.keys():
+        raise ValueError(
+            "Translation must be of: {0}".format(translations.keys()))
+
+    offsets = translations[translation]
+    new_ind = np.copy(ind)
+    test_poly = s_poly(new_ind, dest_grid.nodes)
+    not_found = np.where(~points_in_polys(points, test_polys))[0]
+    for offset in offsets:
+        # for every not found, update the cell to be checked
+        test_polys[not_found] = s_poly(
+            new_ind[not_found] + offset, dest_grid.nodes)
+        # retest the missing points. Some might be found, and will not appear
+        # in still_not_found
+        still_not_found = np.where(
+            ~points_in_polys(points[not_found], test_polys[not_found]))[0]
+        # therefore the points that were found is the intersection of the two
+        found = np.setdiff1d(not_found, still_not_found)
+        # update the indices of the ones that were found
+        not_found = still_not_found
+        new_ind[found] += offset
+        if len(not_found) == 0:
+            break
+
+    # There aren't any boundary issues thanks to numpy's indexing
+    return new_ind
+
+
+# @profile
+def points_in_polys(points, polys, polyy=None):
+    '''
+    :param points: Numpy array of Nx2 points
+    :param polys: Numpy array of N polygons of degree M represented by Mx2 points (NxMx2)
+    for each point, see if respective poly contains it. Returns array of True/False
+    '''
+
+    result = np.zeros((points.shape[0],), dtype=bool)
+    pointsx = points[:, 0]
+    pointsy = points[:, 1]
+    v1x = v1y = v2x = v2y = -1
+    for i in range(0, polys.shape[1]):
+        if polyy is not None:
+            v1x = polys[:, i - 1]
+            v1y = polyy[:, i - 1]
+            v2x = polys[:, i]
+            v2y = polyy[:, i]
+        else:
+            v1x = polys[:, i - 1, 0]
+            v1y = polys[:, i - 1, 1]
+            v2x = polys[:, i, 0]
+            v2y = polys[:, i, 1]
+        test1 = (v2y > pointsy) != (v1y > pointsy)
+        test2 = np.zeros(points.shape[0], dtype=bool)
+        m = np.where(test1 == True)[0]
+        test2[m] = pointsx[m] < \
+            (v1x[m] - v2x[m]) * (pointsy[m] - v2y[m]) / \
+            (v1y[m] - v2y[m]) + v2x[m]
+        np.logical_and(test1, test2, test1)
+        np.logical_xor(result, test1, result)
+    return result
+'''
```

Comment:
This comment will be rendered as a docstring in automated docs (a PR I am preparing). Can you elaborate that into a real comment or even better, an issue?